tests/test_conversion.py

```diff
@@ -0,0 +1,238 @@
+import os
+import sys
+from contextlib import ContextDecorator
+from functools import wraps
+from unittest.mock import patch
+
+import deepspeed
+import torch
+torch.backends.cudnn.deterministic = True
+from transformers import GPT2LMHeadModel, GPT2Tokenizer
+
+from megatron import get_args, get_tokenizer, global_vars, initialize_megatron
+from megatron.testing_utils import (
+    TestCasePlus,
+    mockenv_context,
+    set_seed,
+    torch_assert_equal,
+    torch_assert_close
+)
+from megatron.training import setup_model_and_optimizer
+from megatron.checkpointing import load_checkpoint, save_checkpoint
+
+from pretrain_gpt import get_batch_pipe, model_provider
+from pretrain_gpt import model_provider as gpt_model_provider, get_batch_pipe as get_gpt_batch_pipe
+import json
+
+# fix the relative path
+sys.path.append('tools/convert_checkpoint/')
+from tools.convert_checkpoint import deepspeed_to_transformers, deepspeed_to_megatron
+from test_model import flatten_arguments, get_default_args
+
+
+def reset_global_variables():
+    global_vars._GLOBAL_ARGS = None
+    global_vars._GLOBAL_NUM_MICROBATCHES_CALCULATOR = None
+    global_vars._GLOBAL_TOKENIZER = None
+    global_vars._GLOBAL_TENSORBOARD_WRITER = None
+    global_vars._GLOBAL_ADLR_AUTORESUME = None
+    global_vars._GLOBAL_TIMERS = None
+
+
+class TestConversion(TestCasePlus):
```

Comment:
```suggestion
class TestCheckpointConversion(TestCasePlus):
```
And the same for the filename: e.g. renaming it to a matching `test_checkpoint_conversion.py`

As the current naming is ambiguous.